### PR TITLE
Extract from Patagonia

### LINKED ISSUE #100: Add patagonia product scraping
**Todo**
- [x] Connect to the Patagonia website with beautiful soup
- [x] Locate and extract product code and name

**Story**
As a user, I would like to track the details of products from the Patagonia website.

### PR DESCRIPTION
**Pull request by danfh00**
Adds a script for scraping product code and product name from the Patagonia website.
Closes #100

### CODE REVIEW

File: Dashboard/utilities/extract_from_patagonia.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import json
+import logging
+from bs4 import BeautifulSoup
+from helpers import get_soup, configure_logging
+
+
+def is_correct_page(soup: BeautifulSoup) -> bool:
+    """Returns True if patagonia product page, else False."""
+    if not isinstance(soup, BeautifulSoup):
+        raise TypeError('Soup must be of type BeautifulSoup')
```
Comment (NabihaMoh) [inline Dashboard/utilities/extract_from_patagonia.py]: Should there be logging here?

File: Dashboard/utilities/extract_from_patagonia.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    single_product_identifier = soup.find(
+        'div', attrs={'class': 'product-detail'})
+    return single_product_identifier is not None
+
+
+def scrape_product_information(soup: BeautifulSoup) -> dict | None:
+    """Extract product information from a BeautifulSoup object."""
+    if not isinstance(soup, BeautifulSoup):
+        raise TypeError('Soup must be of type BeautifulSoup')
```
Comment (NabihaMoh) [inline Dashboard/utilities/extract_from_patagonia.py]: Add some logging here too.

File: Dashboard/utilities/extract_from_patagonia.py (hunk 1 above)
Comment (danfh00) [reply Dashboard/utilities/extract_from_patagonia.py]: Thanks, added

File: Dashboard/utilities/extract_from_patagonia.py (hunk 2 above)
Comment (danfh00) [reply Dashboard/utilities/extract_from_patagonia.py]: Also added

**Review by NabihaMoh — APPROVED** [review]
Looks good!

**Pull request merged**